chore(docs): added falco.yaml section about config keys maturity.

FedeDP opened the pull request:
**What type of PR is this?**

/kind cleanup

**Any specific area of the project related to this PR?**

**What this PR does / why we need it**:

Also, rename `Experimental` -> `Incubating` and move `prometheus_metrics_enabled` to `Incubating`.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

See the proposal: https://github.com/falcosecurity/falco/blob/master/proposals/20231220-features-adoption-and-deprecation.md 

**Does this PR introduce a user-facing change?**:

<!--
If NO, just write "NONE" in the release-note block below.

If YES, a release note is required, enter your release note in the block below. 
The convention is the same as for commit messages: https://github.com/falcosecurity/.github/blob/main/CONTRIBUTING.md#commit-convention
If the PR introduces non-backward compatible changes, please add a line starting with "BREAKING CHANGE:" and describe what changed.
For example, `BREAKING CHANGE: the API interface of the rule engine has changed`.
Your note will be included in the changelog.
-->

```release-note
chore(docs): apply features adoption and deprecation proposal to config file keys
```

FedeDP commented:
/milestone 0.38.0

poiana commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/falcosecurity/falco/pull/3206#" title="Author self-approved">FedeDP</a>*

The full list of commands accepted by this bot can be found [here](https://prow.falco.org/command-help?repo=falcosecurity%2Ffalco).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/falcosecurity/falco/blob/master/OWNERS)~~ [FedeDP]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->

FedeDP commented:
/cc @leogr @incertum @LucaGuerra

FedeDP commented on falco.yaml:
```diff
@@ -749,6 +762,8 @@ webserver:
   # Can be an IPV4 or IPV6 address, defaults to IPV4
   listen_address: 0.0.0.0
   k8s_healthz_endpoint: /healthz
+  # [Incubating] `prometheus_metrics_enabled`
```
Am not sure about how to integrate the feature level label here, for a sub-option of a `Stable` config key.

FedeDP commented on falco.yaml:
```diff
@@ -525,7 +538,7 @@ json_include_tags_property: true
 buffered_outputs: false
 
-# [Experimental] `rule_matching`
+# [Incubating] `rule_matching`
```
Experimental -> Incubating

incertum replied:
urgh ok here we should maybe also introduce a synonym `rules_matching` @leogr?

FedeDP replied:
I think singular is ok, in this case; it is `rule-matching algorithm`, feels ok `rule_matching: foo` to me.

FedeDP commented:
I was also wondering whether we could improve somehow the way maturity labels are attached, ie: perhaps moving them to the index at the top? Or moving them just the line before the configuration key, like:
```
# `configs_files`
#
# Falco will load additional configs files specified here.
# Their loading is assumed to be made *after* main config file has been processed,
# exactly in the order they are specified.
# Therefore, loaded config files *can* override values from main config file.
# Also, nested include is not allowed, ie: included config files won't be able to include other config files.
#
# Like for 'rules_files', specifying a folder will load all the configs files present in it in a lexicographical order.
configs_files:
#
# [Stable] 
  - /etc/falco/config.d
```

FedeDP commented on falco.yaml:
```diff
@@ -25,7 +25,7 @@
 #     (Falco command-line arguments)
 #     (Falco environment variables)
-# Falco config files
+# Falco configs files
```
Am not sure though, since we have `watch_config_files` too, and indeed we have multiple configuration (singular) files (plural), so `config_files` is not that bad after all, even if it somehow clashes with `rules_files` (but in this case, we have multiple files with multiple rules inside).
WDYT? Ie i am proposing to rename `configs_files` to `config_files`.
cc @leogr

incertum replied:
So we keep `config` (singular form)? Works for me. Could we re-audit if we are consistent everywhere?

FedeDP replied:
Ok Leo is on board with changing `configs_files` to `config_files`. I will add a new commit for that.

incertum commented on falco.yaml:
```diff
@@ -1170,7 +1185,7 @@ base_syscalls:
 ##############
 
-# [Experimental] `falco_libs` - Potentially subject to more frequent changes
+# [Incubating] `falco_libs` - Potentially subject to more frequent changes
```
Agreed, remove it.

incertum commented:
> I was also wondering whether we could improve somehow the way maturity labels are attached, ie: perhaps moving them to the index at the top? Or moving them just the line before the configuration key, like:
> 
> ```
> # `configs_files`
> #
> # Falco will load additional configs files specified here.
> # Their loading is assumed to be made *after* main config file has been processed,
> # exactly in the order they are specified.
> # Therefore, loaded config files *can* override values from main config file.
> # Also, nested include is not allowed, ie: included config files won't be able to include other config files.
> #
> # Like for 'rules_files', specifying a folder will load all the configs files present in it in a lexicographical order.
> #
> # [Stable] 
> configs_files:
>   - /etc/falco/config.d
> ```

@leogr we haven't responded to this yet. 

- Adding also to the index at the top, big +1
- Keeping `# [Stable] configs_files` was good would say, we can also add to the specific sub-config if needed. Don't have a strong opinion here. You can decide.

FedeDP commented:
Should've done everything:
* `configs_files` -> `config_files`
* `experimental` -> `incubating`
* removed `- Potentially subject to more frequent changes` as @leogr suggested
* Added maturity level to config index
* merged under same section `config_files` and `watch_config_files` settings

FedeDP commented on falco.yaml:
```diff
@@ -25,50 +25,64 @@
-#     watch_config_files
+#     load_plugins [Stable]
+#     plugins [Stable]
 # Falco outputs settings
```
Note for eg: 1.0.0: we might add an `outputs.` namespace for these settings.
Similarly, for all the others, eg:
* `services.{grpc,webserver}`
* log.{stderr,syslog,level,libs}`
...

incertum reviewed (approved):
/approve

poiana commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/falcosecurity/falco/pull/3206#" title="Author self-approved">FedeDP</a>*, *<a href="https://github.com/falcosecurity/falco/pull/3206#pullrequestreview-2066179351" title="Approved">incertum</a>*

The full list of commands accepted by this bot can be found [here](https://prow.falco.org/command-help?repo=falcosecurity%2Ffalco).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/falcosecurity/falco/blob/master/OWNERS)~~ [FedeDP,incertum]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->

leogr reviewed (approved):
cc @LucaGuerra pls keep track of this, it may be useful in release comms.

poiana commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/falcosecurity/falco/pull/3206#" title="Author self-approved">FedeDP</a>*, *<a href="https://github.com/falcosecurity/falco/pull/3206#pullrequestreview-2066179351" title="Approved">incertum</a>*, *<a href="https://github.com/falcosecurity/falco/pull/3206#pullrequestreview-2066272348" title="Approved">leogr</a>*

The full list of commands accepted by this bot can be found [here](https://prow.falco.org/command-help?repo=falcosecurity%2Ffalco).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/falcosecurity/falco/blob/master/OWNERS)~~ [FedeDP,incertum,leogr]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->
